src/utils/decorators/tooltip-decorator/tooltip-decorator.js

```diff
@@ -206,6 +194,28 @@ const TooltipDecorator = (ComposedComponent) => {
       styleElement(tooltip, 'left', `${leftPosition}px`);
     };
 
+    isTooltipOffScreenRight = (position, alignment, tooltipWidth, shifts) => {
+      let pointerX;
+
+      if (position === 'right') {
+        pointerX = shifts.tooltipDistances.right;
+      } else {
```

Comment:
Don't we need a condition here like this?
```js
} else if (position === 'left') {
  return false;
} else {
```
Otherwise, when the position is `'left'`, then `pointerX = shifts.vertical[alignment]` may evaluate to `undefined`.

(When the position is `'left'`, then `alignment` will be one of `'top'`,`'center'`,`'bottom'`, but `shifts.vertical['top']` and `shifts.vertical['bottom']` will evaluate to `undefined`.)

Reply:
and the return value will be evaluated as false, so that condition is not required here

Reply:
Looking at this code in `isTooltipOffScreenRight()`:

```js
  pointerX = shifts.vertical[alignment];
}

return window.innerWidth < pointerX + tooltipWidth;
```

When `shifts.vertical['top']` or `shifts.vertical['bottom']` evaluates to `undefined`, then `pointerX` will have the value `undefined`.

`tooltipWidth` is a `number`, so `pointerX + tooltipWidth` will be `undefined + number`, which evaluates to [`NaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN).

`window.innerWidth < pointerX + tooltipWidth` will then be `number < NaN`.

This does evaluate to `false` for the return value, but I don't think it's good practice to be intentionally allowing `NaN` to be used in this confusing way :warning:

By adding this condition:

```js
} else if (position === 'left') {
  return false;
} else {
```

we avoid the use of `NaN`, and it makes it very clear for future developers about the behaviour of the code.

Reply:
sounds reasonable, added a separate condition for the left position.

Reply:
Many thanks for making this change in ea0c727 :white_check_mark: